### DISCUSSION TITLE
Question creation, recording answers and submitting question set for content state update

### DISCUSSION THREAD

**Discussion by Debayanpaul1593** [Sunbird-inQuiry/Community#96] [Q&A]
Hi Team, 
I had a few doubts with respect to the mentioned topics
1. I need a sample request body for creating a multiple choice question object with question body and answer? I am facing some issues with running the question set editor in local deployment.
2. How are we recording the answers to questions and calculating score from frontend? Could I get some reference to the code
3. When submitting answers for questions we are using `{{sunbird_course_service}}/v1/content/state/update` for updating the answers to questions. We have to send an event for the particular assessment with answer details if I am not mistaken. Where can I get a sample request with this event in the request body?

Thank you

**Comment by rajnishdargan** [comment]
Hi @Debayanpaul1593 
Here is the sample request for creating MCQ question using create question API - [Create Question API](https://inquiry.sunbird.org/learn/product-and-developer-guide/question-and-question-set-service/apis/v2/question-apis/create-question)
What is the issue you are facing in setting up question set editor locally, Please let us know.

The score calculation is happening in player, you can refer the player repo - [QuML Player](https://github.com/Sunbird-inQuiry/player)
One of the score calculation method is [getScore](https://github.com/Sunbird-inQuiry/player/blob/main/projects/quml-library/src/lib/section-player/section-player.component.ts#L889)

[Content state update](https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/apis#course-v1-content-state-update) is part of sunbird lern.
If you have any question specific to learn service you can reach out to [Lern Discussion Forum](https://github.com/orgs/Sunbird-Lern/discussions)

**Comment by Debayanpaul1593** [comment]
Hi @rajnishdargan 
In the [MCQ full request body](https://inquiry.sunbird.org/learn/product-and-developer-guide/question-and-question-set-service/apis/v2/question-apis/create-question#full-request-example) there is a data point given as **outcomeDeclaration** as follows
```JSON
"outcomeDeclaration": {
      "maxScore": {
        "cardinality": "single",
        "type": "integer",
        "defaultValue": 1
      }
    }
```
This data point is not stored in **question_store.question_data** but is stored in **hierarchy_store.questionset_hierarchy**  and is mentioned as external property in schema file of questionset but not of question. Then why are we passing it in the question creation request body? It is being used in getScore calculation for a particular question so how are we fetching that data?

**Reply by rajnishdargan** [reply]
Hi @Debayanpaul1593 
Which version of editor and player you are using?
Could you please provide details on which version of editor, player and API(v1 or v2) you are using.
cc: @sajeshkayyath @krgauraw
